stdlib/public/TensorFlow/CompilerRuntime.swift

```diff
@@ -1259,6 +1244,54 @@ fileprivate func setAttrShapeList(
   }
 }
 
+/// Stack of devices that models nested calls to withDevice/withDefaultDevice.
+/// Devices are represented by their names in TensorFlow notation.
+/// See documentation for `withDevice(_:perform:)` to learn about device names.
+///
+/// All TensorFlow operations will be put on the topmost device on the stack.
+/// When the stack is empty or the topmost device is `nil`, that allows
+/// TensorFlow to place operations on any device that it sees fit.
+@usableFromInline
+class _ThreadLocalState {
+  var deviceScopes: [String?] = []
+
+  private static let key: pthread_key_t = {
+    var key = pthread_key_t()
+    pthread_key_create(&key) {
+#if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
+      let _: AnyObject = Unmanaged.fromOpaque($0).takeRetainedValue()
+#else
+      let _: AnyObject = Unmanaged.fromOpaque($0!).takeRetainedValue()
+#endif
+    }
+    return key
+  }()
+
+  var _currentDevice: String? {
+    return deviceScopes.last ?? nil
+  }
+
+  @usableFromInline
+  func pushDevice(_ device: String?) {
+    deviceScopes.append(device)
+  }
+
+  @usableFromInline
+  func popDevice() {
+    internalConsistencyCheck(deviceScopes.popLast() != nil)
+  }
+
+  @usableFromInline
+  static var value: _ThreadLocalState {
```

Comment:
It is more common for singleton-like type properties like this to be called `local` in Swift APIs.

Also, `_ThreadLocalState` is mechanically the same thing as [ContextManager](https://github.com/tensorflow/swift-apis/blob/e5a288b7ce561fe1b18757d1a440c26eb0c7172d/Sources/DeepLearning/Context.swift#L114) in swift-apis. When we move the TF library to swift-apis, device placement info can be unified into `ContextManager`.

Reply:
Done.